File: src/brs/http/GetMyPeerInfo.java
package brs.http;

import brs.BlockchainProcessor;
import brs.TransactionProcessor;
import javax.servlet.http.HttpServletRequest;
import org.json.simple.JSONObject;
import org.json.simple.JSONStreamAware;

public final class GetMyPeerInfo extends APIServlet.APIRequestHandler {

  private final BlockchainProcessor blockchainProcessor;
  private final TransactionProcessor transactionProcessor;

  public GetMyPeerInfo(BlockchainProcessor blockchainProcessor, TransactionProcessor transactionProcessor) {
    super(new APITag[]{APITag.PEER_INFO});
    this.blockchainProcessor = blockchainProcessor;
    this.transactionProcessor = transactionProcessor;
  }

  @Override
  JSONStreamAware processRequest(HttpServletRequest req) {

    JSONObject response = new JSONObject();
    response.put("walletTTSD", blockchainProcessor.getWalletTTSD());
    response.put("utsInStore", transactionProcessor.getAmountUnconfirmedTransactions());
    return response;
  }

}

File: src/main/java/brs/http/GetMyPeerInfo.kt
package brs.http

import brs.TransactionProcessor
import com.google.gson.JsonElement
import com.google.gson.JsonObject

import javax.servlet.http.HttpServletRequest

internal class GetMyPeerInfo(private val transactionProcessor: TransactionProcessor) : APIServlet.JsonRequestHandler(arrayOf(APITag.PEER_INFO)) {
    override suspend fun processRequest(request: HttpServletRequest): JsonElement {
        val response = JsonObject()
        response.addProperty("utsInStore", transactionProcessor.amountUnconfirmedTransactions)
        return response
    }
}
